gwu-libraries/TweetSets
# Create full extracts as needed via the UI or at command-line

To set up:
- add a full_datasets directory to tweetset_data as described in the README
- in your `docker-compose.yml`, add the volume for the full_dataset into the containers for server, server-flaskrun, and worker, as in `example.cluster-primary.docker-compose.yml`

Testing areas:
- Create an extract on the command-line and confirm a directory is created in full_datasets with the CSV, JSON, and id files. 
- In the UI, select a dataset and then do not add any parameters to the parameters form. Confirm that the files get created in the dataset exports tab, if they did not already exist. If they did already exist, confirm that you see the appropriate files for your IP range. (Change the IP_RANGE definition in .env to simulate a non-GW user)
- Confirm that a custom extract, not the full extract, is created if parameters are used. 
- Confirm that the content on the preview tab is accurate when selecting a full dataset (both by clicking the `Preview` button and by using the tab.
